src/ProxyClient/HttpDispatcher.php

```diff
@@ -104,15 +105,19 @@ public function __construct(
         $this->uriFactory = $uriFactory ?: UriFactoryDiscovery::find();
 
         $this->setServers($servers);
-        $this->setBaseUri($baseUri);
+
+        // Support both, a string or an array of strings (array_filter to kill empty base URLs)
```

Comment:
i am not happy with the array_filter, as that could hide problems. i'd explicitly check `'' === $baseUri` and in that case create [].

Reply:
Done in 5f160db1d7059137fabb1a709e9b6fa36c5939c1. Do you like that better?

Reply:
yep. added another nitpick, but this looks more robust